src/evaluation.py

```diff
@@ -505,9 +538,10 @@ def conditionally_add_explanation(
                 if item is not None
             ]
             self.explanation_table.add_row(*row)
+            self.explanation_table_data_for_csv.append(row)
```

Comment:
Instead of appending the row here, can you check if `self.explanation_table.rows can be looped upon later? It will reduce the coupling.